tokens/style-dictionary.js

```diff
@@ -65,165 +80,188 @@ const colorTransform = (token, theme) => {
  * 2. 'theme' to output only theme's variables (e.g, 'light' or 'dark'), if theme is not provided - only
  * core tokens are built.
  */
-const createCustomCSSVariables = ({
+const createCustomCSSVariables = async ({
   formatterArgs,
   themeVariant,
-}) => {
+}, StyleDictionary) => {
+  // eslint-disable-next-line import/no-unresolved
+  const { sortByReference, usesReferences, getReferences } = (await import('style-dictionary/utils'));
   const { dictionary, options, file } = formatterArgs;
-
   const outputTokens = themeVariant
     ? dictionary.allTokens.filter(token => token.filePath.includes(themeVariant))
     : dictionary.allTokens;
 
   const variables = outputTokens.sort(sortByReference(dictionary)).map(token => {
```

Comment:
Related, upon further investigation, it looks we _do_ rely on token-specific handling of `outputReferences`. For example, we opt-out of including references even when the formatter has them opted-in (e.g., for the case of always including HEX values in SVG `url` values).

That said, token-specific `outputReferences` is still supported with the out-of-the-box `formattedVariables`, by passing `outputReferences` a function instead of a boolean, e.g.:

```jsx
const variables = formattedVariables({
  format: 'css',
  dictionary,
  outputReferences: (token) => {
    // Formatter options suggest to never output references
    if (!outputReferences) {
      return false;
    }
    // Formatter options suggest to output references, but individual token might opt-out
    if (!token.outputReferences) {
      return false;
    }
    // Otherwise, output references.
    return true;
  },
  usesDtcg: true,
});
```

Reply:
[inform] Implemented in follow-up PR: https://github.com/openedx/paragon/pull/3203